internal/rpc/user_devices_test.go

```diff
@@ -250,3 +253,113 @@ func TestGetUserDevice_NoSyntheticDeviceFields_WhenNoTokenID(t *testing.T) {
 
 	assert.Nil(udResult.SyntheticDevice)
 }
+
+func TestClearMetaTransactionRequests(t *testing.T) {
+	assert := assert.New(t)
+	ctx := context.Background()
+	pdb, container := test.StartContainerDatabase(ctx, t, migrationsDirRelPath)
+	defer func() {
+		if err := container.Terminate(ctx); err != nil {
+			t.Fatal(err)
+		}
+	}()
+
+	mtID := ksuid.New().String()
+
+	currTime := time.Now()
+	fifteenminsAgo := currTime.Add(-time.Minute * 15)
+	metaTx := models.MetaTransactionRequest{
+		ID:        mtID,
+		Status:    models.MetaTransactionRequestStatusConfirmed,
+		CreatedAt: fifteenminsAgo,
+	}
+
+	err := metaTx.Insert(ctx, pdb.DBS().Writer, boil.Infer())
+	assert.NoError(err)
+
+	logger := zerolog.Logger{}
+	userDeviceSvc := services.NewUserDeviceService(nil, logger, pdb.DBS, nil, nil)
+	udService := NewUserDeviceRPCService(pdb.DBS, nil, nil, nil, nil, nil, nil, userDeviceSvc)
+
+	resp, err := udService.ClearMetaTransactionRequests(ctx, nil)
+	assert.NoError(err)
+
+	assert.Equal(mtID, resp.Id)
+
+	_, err = models.MetaTransactionRequests(models.MetaTransactionRequestWhere.ID.EQ(mtID)).One(ctx, pdb.DBS().Reader)
+	assert.ErrorIs(err, sql.ErrNoRows)
```

Comment:
Does this pass? we are not wrapping the error when the err == sql.ErrNoRows

Reply:
Yes it does, sqlboiler throws a `sql.ErrNoRows`when there are no records matching the condition. What i'm doing here is retrieving the record directly using an sqlboiler model, its not a custom function so no wrapping.

Reply:
Ahh yes, my confusion I misread the function call above and thought it was `ClearMetaTransactionRequests`